qiskit_optimization/problems/quadratic_program.py

```diff
@@ -811,161 +799,58 @@ def maximize(self,
         self._objective = QuadraticObjective(self, constant, linear, quadratic,
                                              QuadraticObjective.Sense.MAXIMIZE)
 
-    def from_docplex(self, model: Model) -> None:
-        """Loads this quadratic program from a docplex model.
+    def from_model(self, model: Model, translator: Optional[ModelTranslator] = None) -> None:
+        """Loads this quadratic program from an optimization model.
 
         Note that this supports only basic functions of docplex as follows:
         - quadratic objective function
         - linear / quadratic constraints
         - binary / integer / continuous variables
 
         Args:
-            model: The docplex model to be loaded.
+            model: The optimization model to be loaded.
+            translator: The model translators
 
         Raises:
             QiskitOptimizationError: if the model contains unsupported elements.
         """
+        if translator is None:
+            from qiskit_optimization.translators.docplex import DocplexTranslator
+            translator = DocplexTranslator()
+        translator.model_to_qp(model, self)
```

Comment:
Should we have a translator instance explicitly instead of defaulting to DOcplex?